scripts/terminus-installer.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/* Terminus Installer Script
+Created by Alex Fornuto - alex@fornuto.com
+*/
+
+// Define environment variables.
+$pathUpdated = false;
+$paths = explode(":", getenv('PATH'));             // Creates an array with all paths in $PATH
+$installdir = ("/usr/local/bin");                  // Creates a string with the desired installation path
+$rcfiles = array(                                  // Array of common .rc files to look for.
+  ".bashrc",
+  ".zshrc",
+  ".config/fish/config.fish",
+  ".profile",
+);
+$package = "terminus";
+
+// Function to download Terminus executable file from GitHub to /tmp/ then move it to $installdir
+// prompts for sudo if required.
+function downloadTerminus($installdir, $package)
+{
+    // $opts defines values required by the GitHub API to respond correctly. $context formats them for use.
+    $opts = [
+            'http' => [
+              'method' => 'GET',
+              'header' => [
+                'User-Agent: PHP'
+              ]
+            ]
+    ];
+    $context  = stream_context_create($opts);
+    $releases = file_get_contents("https://api.github.com/repos/pantheon-systems/" . $package . "/releases", false, $context);
+    $releases = json_decode($releases);
+    $version  = $releases[0]->tag_name;
+    $url      = $releases[0]->assets[0]->browser_download_url;
+    // Do the needful
+    echo("\nDownloading Terminus " . $version . " from " . $url . " to /tmp \n");
+    $couldDownload = file_put_contents("/tmp/" . $package . ".phar", file_get_contents($url));
+    echo("Moving to " . $installdir . "...\n");
+    if(!rename("/tmp/" . $package . ".phar", $installdir . "/" . $package)){
+        echo("\n" . $installdir . " requires admin rights to write to...\n");
+        exec("sudo mv /tmp/" . $package . ".phar " . $installdir . "/" . $package);
+        echo("\n");
```

Comment:
Is a rename failure also possible if the directory already exists?

Reply:
I don't understand the question. This renames the terminus binary we downloaded into tmp, and then moves into the install directory. No directories are being renamed, and `mv` will overwrite the binary if it already exists in the target directory.

Reply:
It sounds like mv overwrites the binary and won't fail if there's something w/that name already in the directory. OK, good to go then - thanks.